gr-uhd/include/gnuradio/uhd/rfnoc_rx_streamer.h

```diff
@@ -45,12 +45,16 @@ class GR_UHD_API rfnoc_rx_streamer : virtual public gr::sync_block
      * \param vlen Vector length
      * \param issue_stream_cmd_on_start If true, the streamer sends a stream
      *                                  command upstream.
+     * \param start_time_set If true, set start time spec to the stream command
+     * \param start_time The time spec for the stream command if start_time_set is true
      */
     static sptr make(rfnoc_graph::sptr graph,
                      const size_t num_chans,
                      const ::uhd::stream_args_t& stream_args,
                      const size_t vlen = 1,
-                     const bool issue_stream_cmd_on_start = true);
+                     const bool issue_stream_cmd_on_start = true,
+                     const bool start_time_set = false,
```

Comment:
Can we make this redundant? In UHD, a time of 0 often means 'ASAP' (see here: https://github.com/EttusResearch/uhd/blob/master/host/include/uhd/types/time_spec.hpp#L33-L34). Or maybe a `boost::optional`?

Reply:
The [`::uhd::stream_cmd_t`](https://files.ettus.com/manual/structuhd_1_1stream__cmd__t.html) has two fields: `stream_now` and `time_spec`, and they are used both inside the [`start()`](https://github.com/mbr0wn/gnuradio/blob/70ec6a0877f01b248babe437044b7c90bdf4833c/gr-uhd/lib/rfnoc_rx_streamer_impl.cc#L82-L90) method depending on the `d_start_time_set` boolean. Actually, you @mbr0wn wrote this piece of code yourself 3 years ago - I guess you had your reasons to do it like this, and not using a time of 0?   
When you say "0 often means 'ASAP'", the "often" does not sound convincing to be honest ;)  
But I don't know, you are the expert ;)

Reply:
BTW the [`usrp_source`](https://github.com/mbr0wn/gnuradio/blob/70ec6a0877f01b248babe437044b7c90bdf4833c/gr-uhd/lib/usrp_source_impl.cc#L504-L516) never uses the `stream_now`, but **always** uses a "reasonable_delay" of 0.1s, see https://github.com/mbr0wn/gnuradio/blob/70ec6a0877f01b248babe437044b7c90bdf4833c/gr-uhd/lib/usrp_source_impl.cc#L513

Reply:
Hey don't ask me about something I did 3 years ago... or even last week :grin: 

No I think you're right. Having a second look at this (a few days later) I think this is fine.